sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/XmlExternalEntityShouldNotBeParsed_XmlDocument.cs

```diff
@@ -221,6 +221,15 @@ public void Callee(ConfigXmlDocument doc)
         {
             doc.XmlResolver = new XmlPreloadedResolver(); // Noncompliant
         }
-    }
 
+        public void WithInitializeInsideIf()
+        {
+            XmlDocument doc;
+            if ((doc = new XmlDocument()) != null)
```

Comment:
there's a project on Peach which gives a FP here, but I could not repro

I will try to investigate further